cores/esp8266/debug.h

```diff
@@ -5,11 +5,11 @@
 #include <stdint.h>
 
 #ifdef DEBUG_ESP_CORE
-#define DEBUGV(...) ets_printf(__VA_ARGS__)
+#define DEBUGV(fmt, ...) ::printf_P( (PGM_P)PSTR(fmt), ## __VA_ARGS__)
```

Comment:
Should this not be the same as others (i.e. `DEBUG_ESP_PORT.printf(`...)?

Reply:
ets_printf is safe to call in any context (such as system), but DEBUG_ESP_PORT.printf could yield, which would not work in sys context?

Reply:
I changed to `::printf_P` so we can use fmt arguments from flash.
I assume `::printf_P` uses `putchar` installed by `ets_install_putc1/putc2` in debug mode like, `ets_printf` does.
(should the others use `::printf_P` instead of `Serial.printf_P` ?)

Reply:
@igrr Were you suggesting to remove (legacy) `DEBUG_ESP_PORT.printf` and use `DEBUGV` instead ?
About sys context, I believe new newlib's `::vsnprintf()` and fw's `ets_putc()` are safe (no yield).